packages/teleterm/src/ui/types.ts

```diff
@@ -14,39 +14,6 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { MainProcessClient, SubscribeToTshdEvent } from 'teleterm/types';
-import { ClustersService } from 'teleterm/ui/services/clusters';
-import { ModalsService } from 'teleterm/ui/services/modals';
-import { TerminalsService } from 'teleterm/ui/services/terminals';
-import { QuickInputService } from 'teleterm/ui/services/quickInput';
-import { StatePersistenceService } from 'teleterm/ui/services/statePersistence';
-import { CommandLauncher } from 'teleterm/ui/commandLauncher';
-import { KeyboardShortcutsService } from 'teleterm/ui/services/keyboardShortcuts';
-import { WorkspacesService } from 'teleterm/ui/services/workspacesService';
-import { NotificationsService } from 'teleterm/ui/services/notifications';
-import { ConnectionTrackerService } from 'teleterm/ui/services/connectionTracker';
-import { FileTransferService } from 'teleterm/ui/services/fileTransferClient';
-import { ResourcesService } from 'teleterm/ui/services/resources';
-import { ReloginService } from 'teleterm/services/relogin';
-import { TshdNotificationsService } from 'teleterm/services/tshdNotifications';
+import AppContext from 'teleterm/ui/appContext';
 
-export interface IAppContext {
-  clustersService: ClustersService;
-  modalsService: ModalsService;
-  notificationsService: NotificationsService;
-  terminalsService: TerminalsService;
-  keyboardShortcutsService: KeyboardShortcutsService;
-  quickInputService: QuickInputService;
-  statePersistenceService: StatePersistenceService;
-  workspacesService: WorkspacesService;
-  mainProcessClient: MainProcessClient;
-  commandLauncher: CommandLauncher;
-  connectionTracker: ConnectionTrackerService;
-  resourcesService: ResourcesService;
-  fileTransferService: FileTransferService;
-  subscribeToTshdEvent: SubscribeToTshdEvent;
-  reloginService: ReloginService;
-  tshdNotificationsService: TshdNotificationsService;
-
-  init(): Promise<void>;
-}
+export type IAppContext = AppContext;
```

Comment:
>OTOH changing any public field/method of AppContext was super annoying. You had to change AppContext, IAppContext and then the mocks.

Yeah, that annoyed me too, so I thought that removing it was a good decision.

>I found a TS issue on the topic of mocking classes: https://github.com/microsoft/TypeScript/issues/471. What if we followed the idea presented in the last comment? If we added a custom type for this and explained what it's useful for, that could be a good workaround for this problem.

This sounds nice.

Reply:
Now that I think about it I'm not sure if we really need this additional type. If we can just use `IAppContext` then what `PublicOf` gives us? When we add a new field to the context we have two modify both places anyway (I think it was my main motivation :)).
I will revert the change for now.

Reply:
> If we can just use `IAppContext` then what `PublicOf` gives us? When we add a new field to the context we have two modify both places anyway (I think it was my main motivation :)).

Sorry, I wasn't clear enough. If we went with `PublicOf`, then instead of defining both `IAppContext` and `AppContext` we'd only have `AppContext`. Any place which would accept `AppContext` as an arg would need to change the type to `PublicOf<AppContext>` instead. This way there would be only a single place to add a new property.

The downside is that well, we'd need to manually remember about it or we'd need to write a lint rule or something. Which is not really that big of a downside as we have to remember about using `IAppContext` instead of `AppContext` anyway.

But yeah, I'm fine with keeping `IAppContext` for now, even though it's a PITA.